### DISCUSSION TITLE
FT232H

### DISCUSSION THREAD

**Discussion by Norb-23** [nofeletru/UsbAsp-flash#84] [Q&A]
I'm using the FT232H (FTDI) and your programmer does flashing of a 512kb MX25L4006 in about 3 seconds.
I do the programming of only ONE block in not less than 7 seconds...
What is your secret?
I use PP (02h) for page programming and have always to WREN (06h) before writing. 
Can you exceed the page boundaries?
Could you share some source code or functions?

Thank you very much for your efforts!

Regards
Norbert

**Comment by nofeletru** [comment]
You may look at [ft232hhw.pas](https://github.com/nofeletru/UsbAsp-flash/blob/master/software/ft232hhw.pas) to see how i work with SPI and initialize FT232H.
I'm playing around with packing commands in FT232H buffer for faster response.
Also have option to skip pages that have all bytes FF(thus it no sense to write blank pages).
